### PR TITLE
Port DesignerAction*Collection components

### LINKED ISSUE #609: Port various components from framework to core
- [x] System.ComponentModel.Design.DesignerActionItemCollection
 - [x] System.ComponentModel.Design.DesignerActionListCollection
 - [ ] ~~System.ServiceModel.Activation.AspNetCompatibilityRequirementsAttribute~~ This component doesn't belong to our team

Related requests: #221

### PR DESCRIPTION
**Pull request by JuditRose**
Fixes #609 

Components copied from framework.

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionListCollection.cs [hunk 1]
```diff
@@ -21,48 +22,63 @@ public DesignerActionListCollection(DesignerActionList[] value)
...
-            throw new NotImplementedException(SR.NotImplementedByDesign);
+            return List.Add(value);
         }
 
         public void AddRange(DesignerActionList[] value)
         {
-            throw new NotImplementedException(SR.NotImplementedByDesign);
+            if (value == null)
+            {
+                throw new ArgumentNullException("value");
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionListCollection.cs]: >value [](start = 49, length = 5)

please use `nameof`

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionListCollection.cs [hunk 2]
```diff
@@ -21,48 +22,63 @@ public DesignerActionListCollection(DesignerActionList[] value)
...
+                Add(value[i]);
+            }
         }
 
         public void AddRange(DesignerActionListCollection value)
         {
-            throw new NotImplementedException(SR.NotImplementedByDesign);
+            if (value == null)
+            {
+                throw new ArgumentNullException("value");
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionListCollection.cs]: nameof

File: src/System.Windows.Forms.Design/tests/UnitTests/DesignerActionItemCollectionTests.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+            underTest.Add(item1);
+            underTest.Add(item2);
+
+            underTest.Add(item3);
+            Assert.Equal(2, underTest.IndexOf(item3));
+
+            underTest.Insert(2, item4);
+            Assert.Equal(2, underTest.IndexOf(item4));
+
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/tests/UnitTests/DesignerActionItemCollectionTests.cs]: Consider adding a test for index for item3 here.

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionItemCollection.cs [hunk 4]
```diff
@@ -10,38 +10,39 @@ public class DesignerActionItemCollection : CollectionBase
...
-            get => throw new NotImplementedException(SR.NotImplementedByDesign);
-            set => throw new NotImplementedException(SR.NotImplementedByDesign);
+            get => (DesignerActionItem)(List[index]);
+            set => List[index] = value;
         }
 
         public int Add(DesignerActionItem value)
         {
-            throw new NotImplementedException(SR.NotImplementedByDesign);
+            int index = List.Add(value);
```
Comment (hughbe) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionItemCollection.cs]: nit: could do `=> List.Add(value)`

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionListCollection.cs [hunk 5]
```diff
@@ -21,48 +22,63 @@ public DesignerActionListCollection(DesignerActionList[] value)
...
         }
 
         public void AddRange(DesignerActionList[] value)
         {
-            throw new NotImplementedException(SR.NotImplementedByDesign);
+            if (value == null)
+            {
+                throw new ArgumentNullException("value");
+            }
+            for (int i = 0; ((i) < (value.Length)); i = ((i) + (1)))
```
Comment (hughbe) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionListCollection.cs]: Nit: strange bracketing

File: src/System.Windows.Forms.Design/tests/UnitTests/DesignerActionItemCollectionTests.cs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+namespace System.Windows.Forms.Design.Tests
+{
+    public class DesignerActionItemCollectionTests
+    {
+        [Fact]
+        public void DesignerActionItemCollection_Constructor()
+        {
+            DesignerActionItemCollection underTest = new DesignerActionItemCollection();
+            Assert.NotNull(underTest);
```
Comment (hughbe) [inline src/System.Windows.Forms.Design/tests/UnitTests/DesignerActionItemCollectionTests.cs]: Could validate empty?

**Pull request merged**